src/sdam/monitor.rs

```diff
@@ -122,13 +122,15 @@ impl HeartbeatMonitor {
                 None => break,
             };
 
+            // subscribe to check requests before performing the check in case one comes in
+            // after the check completes
+            let mut topology_check_requests_subscriber =
+                topology.subscribe_to_topology_check_requests();
+
             if self.check_server(&topology, &server).await {
                 topology.notify_topology_changed();
             }
 
-            let mut topology_check_requests_subscriber =
```

Comment:
My theory is that the following sequence happens:
- check completes
- monitor notifies that topology changed
- operation sleeping in server selection gets notification and quickly fails server selection, requesting another check
- monitor starts subscribing to request checks, missing the previous one
- monitor sleeps until heartbeatFrequencyMS is hit

This fixes that by ensuring any requests that come in after the check will be recorded too, hopefully fixing this case.